cmd/otel-allocator/watcher/promOperator.go

```diff
@@ -57,7 +57,16 @@ func NewPrometheusCRWatcher(cfg allocatorconfig.Config, cliConfig allocatorconfi
 		monitoringv1.PodMonitorName:     podMonitorInformers,
 	}
 
-	generator, err := prometheus.NewConfigGenerator(log.NewNopLogger(), &monitoringv1.Prometheus{}, true) // TODO replace Nop?
+	// TODO: We should make these durations configurable
+	prom := &monitoringv1.Prometheus{
+		Spec: monitoringv1.PrometheusSpec{
+			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
+				ScrapeInterval: monitoringv1.Duration("30s"),
+			},
+		},
+	}
+
+	generator, err := prometheus.NewConfigGenerator(log.NewNopLogger(), prom, true) // TODO replace Nop?
```

Comment:
```suggestion
	prometheusInstance := &monitoringv1.Prometheus{
		Spec: monitoringv1.PrometheusSpec{
			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
				ScrapeInterval: monitoringv1.Duration(cfg.Config.GlobalConfig.ScrapeInterval.String()),
				ScrapeTimeout:  monitoringv1.Duration(cfg.Config.GlobalConfig.ScrapeTimeout.String()),
			},
		},
	}

	generator, err := prometheus.NewConfigGenerator(log.NewNopLogger(), prometheusInstance, true) // TODO replace Nop?
```

I would suggest to fix several things at once. Setting these params this way we fix the issue and get them from the parsed config at the same, so make the configurable

WDYT?

Reply:
I think this is a good suggestion @achetronic. WDYT @jaronoff97 @swiatekm-sumo? We could then directly address https://github.com/open-telemetry/opentelemetry-operator/pull/1822#discussion_r1226748692. Now I wonder why it was configurable from the beginning. And in case the intervals are empty we would simply fall back to a default.

Reply:
@matej-g 
I was wondering if target allocator needs to be configurable as its responsibility is only scrape ServiceMonitor/PodMonitor resources to create like the half of config related to scrape_configs, right? the rest of the file is created by the operator itself. Am I wrong?

Reply:
I'm a fan of this, however, I think we should add these as fields for the TA rather than use the prometheus configuration's global config. First, some users may not be setting a prometheus configuration for their TA (a valid setup) and it would be a bit confusing to need to set only the global config and nothing else. Second, I think given this is for the TA this would be better served as a CLI flag for the TA that we add support for in the collector's CRD. 

Given ^^ I think we should do that in a follow-up and not couple those changes in with this PR.

Reply:
@jaronoff97 Hey, I'm completely new to this project :) 

Could you describe the approach a bit better? I would like to collaborate to fix this ASAP as it is affecting our setup

Reply:
Hey no worries! Yeah I'd love to get this PR in today. I'm proposing that rather than using the global config which is embedded within the scrape config for the target allocator, we add new fields to the target allocator and use those directly. I prefer this approach because if someone doesn't have a scrape config set they could still define a global scrape config and scrape interval. I'd like to decouple these changes into separate PRs, however, because adding more changes to this one would increase it's scope which would require more reviewing. If you'd like to chat more or even pair on these changes, reach out in our slack channel (#otel-operator) in the CNCF slack :)

Reply:
Hey, thank you for merging :) Of course i will reach some of you in slack